tests/end_to_end.test.js

```diff
@@ -376,5 +376,53 @@ const Pulsar = require('../index.js');
       await reader.close();
       await client.close();
     });
+
+    test('Produce/Read (Compression)', async () => {
+      const client = new Pulsar.Client({
+        serviceUrl: 'pulsar://localhost:6650',
+        operationTimeoutSeconds: 30,
+      });
+      expect(client).not.toBeNull();
+
+      const topic = 'persistent://public/default/produce-read-compression';
+      const producer = await client.createProducer({
+        topic,
+        sendTimeoutMs: 30000,
+        batchingEnabled: true,
+        compressionType: 'ZSTD',
```

Comment:
Could you add a test for SNAPPY as well?

Reply:
Ah, it seems that sending a message fails when the compression type is set to SNAPPY. I think this is because the Docker image for building the DEB package doesn't include the SNAPPY library.
https://github.com/apache/pulsar/blob/master/pulsar-client-cpp/pkg/rpm/Dockerfile

So we shouldn't add a test for SNAPPY yet.

Reply:
I can see if I can update the docker container and include the tests.

Reply:
I don't think SNAPPY is available unless the SNAPPY library is installed when the C++ client DEB package is built.
https://github.com/apache/pulsar/pull/4259/files#diff-bda8db832e446d092d38464123ed17bc
https://github.com/apache/pulsar/pull/4259/files#diff-7e441fcfcb20e339c71f7f8f48569579

So, even if you modify the Docker image, we cannot use SNAPPY immediately. We have to wait for the next release of the C++ package.

Reply:
Created a pull-request to the apache/pulsar repository.
https://github.com/apache/pulsar/pull/8086